sheets/3/exercise-2.tex

```diff
@@ -1,11 +1,107 @@
-    \excercise{Schleifen}
-            \subexcercise Schreibe deine \texttt{turnLeft()} so um, dass sie eine for-Schleife verwendet.
-
-            \subexcercise Neo muss so schnell wie möglich die nächstgelegene Telefonzelle erreichen, um die Matrix zu verlassen. Leider weiß er nicht genau, wie weit diese entfernt ist. Schreibe hierzu eine Operation, die Neo so lange geradeaus laufen lässt, bis die Operation \texttt{Neo.isOnPhoneBooth()} den Wert \texttt{true} zurück gibt. Verwende in deiner Lösung eine while-Schleife.
-
-            \subexcercise(\textbf{Bonus}) Schreibe die Operation aus \textit{(b)} so um, dass sie eine do-while-Schleife verwendet.
-
-            \subexcercise Dieses Mal haben die Agenten alle Telefonzellen in Neos Nähe zerstört. Welche Auswirkung hat das auf deinen Code aus \textit{(b)}? Nutze die Operation \texttt{Neo.canMove()}, um den Fehler zu beheben.
-
-            \textbf{Für diese Aufgabe} müsst ihr \texttt{this.boothsDestroyed} auf \texttt{true} setzen.
-
+\excercise{Schleifen}
+	\subsection*{a)}
+		Passe die Main Methode an die Aufgabe an.
+	\subsection*{b)}
+		Wir möchten wissen wie viele Münzen auf dem Feld sind auf dem sich Neo befindet. Dafür heben wir so lange immer eine Münze auf bis sich keine Münzen mehr auf dem Feld befinden. Wenn die Aufgehobenen Münzen zählen können wir somit 				herausfinden wie viele Münzen auf dem Feld sind.

```

Comment:
Das ist viel zu kompliziert für b), aber das ist mein Fehler. Mit Beispielcode hab ich eher an sowas gedacht:

```java
Integer coinCount = neo.getCurrentlyCollectableEntities(Coin.class, true).size();
```

Aber am besten legt man auf neo direkt eine Methode `getCurrentlyCollectableCoins()` an die die Parameter oben schon vorausfüllt.
In dieser Aufgabe kannst du auch erklären, dass die Liste eine Länge hat die wir mit `.size()` abfragen können (noch nicht mehr zu listen erklären)